autoload/arduino.vim

```diff
@@ -118,19 +121,36 @@ endfunction
 function! arduino#GetBoards()
   let arduino_dir = arduino#GetArduinoDir()
   let boards = []
-  for filename in split(globpath(arduino_dir . '/hardware', '**/boards.txt'), '\n')
-    let pieces = split(filename, '/')
-    let package = pieces[-3]
-    let arch = pieces[-2]
-    let lines = readfile(filename)
-    for line in lines
-      if line =~? '^[^.]*\.build\.board=.*$'
-        let linesplit = split(line, '\.')
-        let board = linesplit[0]
-        call add(boards, package . ':' . arch . ':' . board)
-      endif
+  if arduino#GetArduinoUserInstallation() == 1
+    for filename in split(globpath(arduino_dir . '/packages', '**/boards.txt'), '\n')
```

Comment:
Is the difference in logic here (searching `/packages/` instead of `/hardware`, pulling out the package version, etc) due to the user installation? I would expect that boards might exist in the `/packages` dir even if system installed, and that boards would exist under `/hardware` in a user installation. Is this the case?

Reply:
Yes you got the point correct: I realized with your plugin, that all my hardware, etc. (installed by the arduino ide into the user dir) wasn't available.
That's why I started to dig into the code.